### PR TITLE
basic cue sheet syntax checking

### PR DESCRIPTION
**Pull request by wdlkmpx**
Instead of adding like 7 extra if's (CATALOG, CDTEXTFILE, FILE, FLAGS, ISRC, POSTGAP, REM)
i decided to create an array of strings and check each line against that array..

Logs unknown cue commands as errors. Something like this:
Unknown cue sheet item: COMMENT "AAAA"

### CODE REVIEW

File: playlist.c [hunk 1]
```diff
@@ -1069,6 +1069,12 @@ plt_insert_cue_from_buffer_int (playlist_t *playlist, playItem_t *after, playIte
 
     const char *charset = junk_detect_charset_len (buffer, buffersize);
 
+    char *cue_commands[] = {
```
Comment (Alexey-Yakovenko) [inline playlist.c]: This needs to be declared as `const char *`.

File: playlist.c [hunk 2]
```diff
@@ -1069,6 +1069,12 @@ plt_insert_cue_from_buffer_int (playlist_t *playlist, playItem_t *after, playIte
...
+        "CATALOG ", "CDTEXTFILE ", "FILE ", "FLAGS ", "INDEX ", "ISRC ",
```
Comment (Alexey-Yakovenko) [inline playlist.c]: There's no need to save space and combine multiple values on the same line.
Diff tools work better when the values each take a separate line, so please split them up.

File: playlist.c [hunk 3]
```diff
@@ -1119,6 +1125,20 @@ plt_insert_cue_from_buffer_int (playlist_t *playlist, playItem_t *after, playIte
         buffersize -= p-buffer;
         buffer = p;
         p = pl_cue_skipspaces (str);
+
+        // validate cue command
+        int i = 0, valid_cue_command = 0;
```
Comment (Alexey-Yakovenko) [inline playlist.c]: it doesn't seem like the `i` variable is used anywhere outside of the for loop before, so it should be declared inline: `for (int i = 0; ...`, then this line should look this way: `int valid_cue_command = 0;`

File: playlist.c [hunk 4]
```diff
@@ -1119,6 +1125,20 @@ plt_insert_cue_from_buffer_int (playlist_t *playlist, playItem_t *after, playIte
...
+        for (i = 0; cue_commands[i]; i++) {
+            if (!strncmp (p, cue_commands[i], strlen(cue_commands[i]))) {
+               valid_cue_command = 1;
+               break;
+            }
+        }
+        if (valid_cue_command == 0) {
+            trace_err ("Unknown cue sheet item: %s\n", p);
```
Comment (Alexey-Yakovenko) [inline playlist.c]: This error message needs to be removed, unless it points to the cuesheet file name, or the track filename for embedded cuesheet.

**Comment by Alexey-Yakovenko** [comment]
A few more notes about this change...

* While I can see how you added checking for valid cuesheet commands, I can't see where or whether you added the actual handling for them. E.g. `CDTEXTFILE` is in the list of valid commands, but there's no code which parses and stores it in RAM.

* Throwing an error when invalid command is encountered is quite pointless, when there's no actual error condition. The invalid/unknown command is skipped, and the parser moves on.

**Comment by wdlkmpx** [comment]
I think I better close this and open a new pull request that deals with some meaningful "extra tags" from cue sheets, a more generic method that includes REM DATE, REM GENRE and REM COMMENT

**Comment by Alexey-Yakovenko** [comment]
Ok, feel free to do that.

**Pull request closed without merging**